File: SWA.Ariadne.Logic/RandomBacktracker.cs
using System;
using System.Collections.Generic;
using System.Text;
using SWA.Ariadne.Model;

namespace SWA.Ariadne.Logic
{
    /// <summary>
    /// A MazeSolver with one current path and backtracking.
    /// At a crossing in forward direction: Chooses a random open wall.
    /// </summary>
    internal class RandomBacktracker : BacktrackerBase
    {
        #region Constructor

        /// <summary>
        /// Constructor.
        /// </summary>
        /// <param name="maze"></param>
        /// <param name="mazeDrawer"></param>
        public RandomBacktracker(Maze maze, IMazeDrawer mazeDrawer)
            : base(maze, mazeDrawer)
        {
        }

        #endregion

        #region Runtime methods

        /// <summary>
        /// Select one of the open walls leading away from the given square.
        /// </summary>
        /// <param name="sq1"></param>
        /// <param name="openWalls"></param>
        /// <returns></returns>
        protected override MazeSquare.WallPosition SelectDirection(MazeSquare sq1, List<MazeSquare.WallPosition> openWalls)
        {
            return openWalls[random.Next(openWalls.Count)];
        }

        #endregion
    }
}

File: src.JAVA/src/swa/ariadne/logic/RandomBacktracker.java
package swa.ariadne.logic;

import java.util.List;

import swa.ariadne.model.Maze;
import swa.ariadne.model.MazeSquare;
import swa.ariadne.model.WallPosition;

/**
 * A {@link IMazeSolver MazeSolver} with one current path and backtracking.
 * At a crossing in forward direction: Chooses a random open wall.
 * 
 * @author Stephan.Wacker@web.de
 */
public
class RandomBacktracker
extends BacktrackerBase
{
    /**
     * Constructor.
     * @param maze The problem to be solved.
     * @param mazeDrawer An object that can draw the maze. 
     */
    public RandomBacktracker(Maze maze, IMazeDrawer mazeDrawer)
    {
        super(maze, mazeDrawer);
    }

    @Override
    protected WallPosition selectDirection(MazeSquare sq, List<WallPosition> openWalls) 
    {
        return openWalls.get(random.nextInt(openWalls.size()));
    }

}
